WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift

```diff
@@ -845,20 +854,32 @@ private extension OrderListViewController {
         self.present(surveyNavigation, animated: true, completion: nil)
     }
 
-    private func showIPPFeedbackDismissAlert() {
+    private func showIPPFeedbackDismissAlert(survey: SurveyViewController.Source, campaign: FeatureAnnouncementCampaign ) {
         let actionSheet = UIAlertController(
             title: Localization.dismissTitle,
             message: Localization.dismissMessage,
             preferredStyle: .alert
         )
 
         let remindMeLaterAction = UIAlertAction( title: Localization.remindMeLater, style: .default) { [weak self] _ in
-            self?.viewModel.IPPFeedbackBannerRemindMeLaterTapped()
+            self?.viewModel.IPPFeedbackBannerRemindMeLaterTapped(for: campaign)
+            ServiceLocator.analytics.track(
+                event: .InPersonPaymentsFeedbackBanner.dismissed(
+                    source: .orderList,
+                    campaign: campaign,
+                    remindLater: true)
+            )
         }
         actionSheet.addAction(remindMeLaterAction)
 
         let dontShowAgainAction = UIAlertAction( title: Localization.dontShowAgain, style: .default) { [weak self] _ in
-            self?.viewModel.IPPFeedbackBannerDontShowAgainTapped()
+            self?.viewModel.IPPFeedbackBannerDontShowAgainTapped(for: campaign)
+            ServiceLocator.analytics.track(
+                event: .InPersonPaymentsFeedbackBanner.dismissed(
+                    source: .orderList,
+                    campaign: campaign,
+                    remindLater: false)
+            )
```

Comment:
IMO, instead of firing the analytics events from the view controller, if we move the tracking code to `OrderListViewModel` and inject `analytics`, we could unit test the tracks events and their properties. For this code, we can do the tracking from inside `func IPPFeedbackBannerDontShowAgainTapped(for campaign: FeatureAnnouncementCampaign)` method.

What do you think?

Reply:
Good catch! I moved this to the ViewModel as well as injected Analytics on 6775a27165302f465676225c23bbbfece95159d2 . Added some initial Unit Tests on e7c67ee3b0107c84356ce846f0bed19c2b74f044 . I have opened a different PR to add additional tests if needed [here](https://github.com/woocommerce/woocommerce-ios/issues/8655).

Reply:
Moved the CTA track to the ViewModel too, as well as an initial test on cfe315f6c3124a99cfe7fd05bed167c7e38fb77c